From: Gavin Shan <gshan@redhat.com>
Subject: mm/sparse: expose for_each_present_section_nr()
Date: Tue, 11 Mar 2025 10:46:55 +1000

Patch series "drivers/base/memory: Two cleanups", v2.

The series has two cleanups to drivers/base/memory as below.

PATCH[1-2] Improves add_boot_memory_block() by using for_each_present_section_nr()
           and droping @section_count
PATCH[3]   Corrects the comments to match with 'struct memory_group'


This patch (of 2):

Expose for_each_present_section_nr() to be used by drivers/base/memory
in the next patch.

No functional changes intended.

Link: https://lkml.kernel.org/r/20250311004657.82985-1-gshan@redhat.com
Link: https://lkml.kernel.org/r/20250311004657.82985-2-gshan@redhat.com
Signed-off-by: Gavin Shan <gshan@redhat.com>
Cc: Danilo Krummrich <dakr@kernel.org>
Cc: David Hildenbrand <david@redhat.com>
Cc: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Cc: Oscar Salvador <osalvador@suse.de>
Cc: "Rafael J. Wysocki" <rafael@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/mmzone.h |    5 +++++
 mm/sparse.c            |    5 -----
 2 files changed, 5 insertions(+), 5 deletions(-)

--- a/include/linux/mmzone.h~mm-sparse-expose-for_each_present_section_nr
+++ a/include/linux/mmzone.h
@@ -2140,6 +2140,11 @@ static inline unsigned long next_present
 	return -1;
 }
 
+#define for_each_present_section_nr(start, section_nr)		\
+	for (section_nr = next_present_section_nr(start - 1);	\
+	     section_nr != -1;					\
+	     section_nr = next_present_section_nr(section_nr))
+
 /*
  * These are _only_ used during initialisation, therefore they
  * can use __initdata ...  They could have names to indicate
--- a/mm/sparse.c~mm-sparse-expose-for_each_present_section_nr
+++ a/mm/sparse.c
@@ -170,11 +170,6 @@ static void __section_mark_present(struc
 	ms->section_mem_map |= SECTION_MARKED_PRESENT;
 }
 
-#define for_each_present_section_nr(start, section_nr)		\
-	for (section_nr = next_present_section_nr(start-1);	\
-	     section_nr != -1;								\
-	     section_nr = next_present_section_nr(section_nr))
-
 static inline unsigned long first_present_section_nr(void)
 {
 	return next_present_section_nr(-1);
_
